sql/src/main/java/io/crate/planner/consumer/CrossJoinConsumer.java

```diff
@@ -248,23 +362,33 @@ private NestedLoop toNestedLoop(List<QueriedTableRelation> queriedTables, Consum
                     leftOrderBy = mergedOrderBy(leftRelation, rightRelation);
                     leftPlan = nl;
                 }
-                MergePhase leftMerge = mergePhase(
-                        context,
-                        localExecutionNodes,
-                        leftPlan.resultNode(),
-                        leftOutputs,
-                        leftOrderBy
-                );
+                MergePhase leftMerge = null;
+                if (!leftPlan.resultNode().executionNodes().isEmpty()
+                        && !leftPlan.resultNode().executionNodes().equals(localExecutionNodes)) {
```

Comment:
mergePhase() method is doing exactly the same check and returning null otherwise, pls revert